app/(main)/markets/AiChat.tsx

```diff
@@ -12,65 +12,111 @@ import { Icon, IconButton, Input, Tag } from '@swapr/ui';
+  'The oracle stirs, the pages turn… soon, the answer.',
+  'I will show you the answer! If my internet allows…',
 ];
 
+const randomWaitingMessageIndex = Math.round(Math.random() * waitingMessges.length);
+
+const fetchMessages = async (id: Address) => {
+  const response = await fetch(`${MARKET_CHAT_URL}/api/market-chat?market_id=${id}`);
+  if (!response.ok) {
+    throw new Error('Network response was not ok');
+  }
+  return response.json();
+};
+
 export const AiChat = ({ id }: AiChatProps) => {
+  const [isOpen, setOpen] = useState(false);
   const scrollAreaRef = useRef<HTMLDivElement>(null);
-  const [messages, setMessages] = useState(mockMessages);
+
+  const {
+    data,
+    error: messagesError,
+    isFetched: isFetchedMessages,
+  } = useQuery({
+    queryKey: ['fetchMessages', id],
+    queryFn: () => fetchMessages(id),
+  });
+  const { data: market } = useQuery<Pick<Query, 'fixedProductMarketMaker'>>({
+    queryKey: ['getMarket', id],
+    queryFn: async () => getMarket({ id }),
+    staleTime: Infinity,
+  });
+
+  const {
+    messages,
+    error: chatError,
+    append,
+  } = useChat({
+    id,
+    body: {
+      marketId: id,
+    },
+    api: `${MARKET_CHAT_URL}/api/market-chat`,
+    initialMessages: data,
+  });
+
+  const title = market?.fixedProductMarketMaker?.title;
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    if (signal.aborted) return;
+
+    if (isFetchedMessages && data?.length === 0 && title && isOpen) {
+      append({ role: 'user', content: title });
+    }
+
+    return () => controller.abort();
+  }, [append, data?.length, isFetchedMessages, isOpen, title]);
 
   useEffect(() => {
     if (scrollAreaRef.current) {
       scrollAreaRef.current.scrollTop = scrollAreaRef.current.scrollHeight;
     }
   }, [scrollAreaRef]);
 
+  const assistantMessages = messages.filter(({ role }) => role === 'assistant');
+  const noAssistantMessages =
+    messages.length === 1 &&
+    (assistantMessages.length === 0 || assistantMessages?.at(0)?.content.length === 0);
+  const hasError = messagesError || chatError;
+
   return (
-    <Dialog.Root modal>
+    <Dialog.Root modal onOpenChange={isOpen => setOpen(isOpen)}>
       <div className="fixed bottom-10 flex w-full items-center justify-end px-2 md:px-6">
         <div className="flex flex-col items-end">
           <Dialog.Portal>
-            <Dialog.Content
-              className="data-[state=open]:animate-grow fixed bottom-28 right-0 w-full origin-bottom-right rounded-16 border border-outline-base-em bg-surface-surface-0 shadow-2 md:right-4 md:w-[420px]"
-              onInteractOutside={e => {
-                e.preventDefault();
-              }}
-            >
+            <Dialog.Content className="fixed bottom-28 right-0 w-full origin-bottom-right rounded-16 border border-outline-base-em bg-surface-surface-0 shadow-2 data-[state=open]:animate-grow md:right-4 md:w-[420px]">
```

Comment:
We're getting a warning about a missing prop for the DialogContent. I didn't see an element in this file that could nicely describe the Dialog content so another valid value is `undefined` 
```suggestion
            <Dialog.Content
              className="fixed bottom-28 right-0 w-full origin-bottom-right rounded-16 border border-outline-base-em bg-surface-surface-0 shadow-2 data-[state=open]:animate-grow md:right-4 md:w-[420px]"
              aria-describedby={undefined}
            >
```

Reply:
Added a description :)